src/composables/useJwt.ts

```diff
@@ -0,0 +1,104 @@
+// libraries
+import { Notify } from 'quasar';
+
+// composables
+import { i18n } from '../boot/i18n';
+
+// types
+interface JwtParts {
+  header: string;
+  payload: string;
+  signature: string;
+}
+
+export const useJwt = () => {
+  /**
+   * Get JWT expiration time
+   * @param {string} token - JWT token
+   * @return {number | null} Expiration time in seconds or null if invalid
+   */
+  const readJwtExpiration = (token: string): number | null => {
+    try {
+      const { payload } = parseJwt(token);
+      const decodedPayload = decodePayload(payload);
+      // extract the expiration time
+      const expirationTime = decodedPayload.exp;
+      if (!expirationTime) {
+        Notify.create({
+          message: i18n.global.t('refreshTokens.messageJwtInvalidExpiration'),
+          color: 'negative',
+        });
+        return null;
+      }
+      return expirationTime as number;
+    } catch (error) {
+      if (error instanceof Error) {
+        Notify.create({
+          message: i18n.global.t('refreshTokens.messageJwtInvalidWithMessage', {
+            error: error.message,
+          }),
+          color: 'negative',
+        });
+      } else {
+        Notify.create({
+          message: i18n.global.t('refreshTokens.messageJwtInvalid'),
+          color: 'negative',
+        });
+      }
+      return null;
+    }
+  };
+
+  /**
+   * Parse JWT token into parts
+   * @param {string} token - JWT token string
+   * @return {JwtParts} Object (header, payload, signature) - token parts
+   * @throws {Error} If the token format is invalid
+   */
+  const parseJwt = (token: string): JwtParts => {
+    const parts = token.split('.');
+    if (parts.length !== 3) {
+      throw new Error('Invalid JWT token format');
```

Comment:
You can translate error message, please.

Reply:
Translated.